generate_symbol_map.py

```
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser
+from elftools.elf.elffile import ELFFile
+from itanium_demangler import parse
+import re
+import sys
+
+argument_parser = ArgumentParser()
+argument_parser.add_argument('elf_file_path')
+argument_parser.add_argument('smap_file_path')
+args = argument_parser.parse_args()
+
+try:
+    elf_file_stream = open(args.elf_file_path, 'rb')
+except:
+    sys.exit(f'Failed to open the file \'{args.elf_file_path}\'!')
+
+symtab_section = ELFFile(elf_file_stream).get_section_by_name('.symtab')
+if symtab_section is None:
+    sys.exit(f'There exists no \'.symtab\' section in the .elf file \'{args.elf_file_path}\'!')
+
+bad_symbol_patterns = [
+    re.compile(r'\.(.*)$'),
+    re.compile(r'^_GLOBAL__sub_[A-Z]_'),
+    re.compile(r'^patch_'),
+    re.compile(r'^replaced_'),
+    re.compile(r'^replacement_'),
+]
+
+dictionary = {}
+for symbol in symtab_section.iter_symbols():
+    if (symbol.entry['st_value'] < 0x80000000):
+        continue
+
+    symbol_name = symbol.name
+    for bad_symbol_pattern in bad_symbol_patterns:
+        symbol_name = bad_symbol_pattern.sub('', symbol_name)
+
+    try:
+        demangled_symbol_name = parse(symbol_name)
+        if demangled_symbol_name:
+            symbol_name = demangled_symbol_name
+    except:
+        pass
+
+    dictionary[symbol.entry['st_value']] = symbol_name
+
+try:
+    smap_file_stream = open(args.smap_file_path, 'w')
+except:
+    sys.exit(f'Failed to open the symbol map file \'{args.smap_file_path}\'!')
+try:
+    for item in sorted(dictionary.items()):
+        smap_file_stream.write(f"0x{item[0]:X} {item[1]}\n")
+except:
+    sys.exit(f'Failed to write to the symbol map file \'{args.smap_file_path}\'!')
```

Using bare except sounds kind of meh, especially when not using `with` for file handling, could this be refactored into a function and errors handled more specifically?

---

> Using bare except sounds kind of meh, especially when not using `with` for file handling

File objects are flushed and closed once a script finishes executing.

> could this be refactored into a function and errors handled more specifically?

Which exceptions should be treated differently ?

---

File objects being flushed and closed when a script closes is probably implementation defined behavior, and I've had issues before with Windows breaking due to this issue. It's mostly just good practice to use `with` and avoid bare `except`s, but specifically `KeyboardException` and maybe other internal stuff.

---

Resolved in commit [088a389](https://github.com/stblr/mkw-sp/pull/717/commits/088a38970d0048d12046d18841b79e2ca45f7432).